Update help docs

nlhkabu opened the pull request:
Closes #2659, Closes #2225 

![screencapture-localhost-help-1513705521862](https://user-images.githubusercontent.com/3323703/34171737-60dbf4a0-e4e7-11e7-9980-c79f4563e494.png)

ewdurbin commented:
Ohhhhh while we're in here @nlhkabu can we add a clickable/copyable anchor link for each `id`?

nlhkabu commented:
@ewdurbin already done :grin:

dstufft commented on warehouse/templates/pages/help.html:
```diff
@@ -36,23 +65,57 @@ <h3 id="packages" class="common-question">What's a package, project, or release?
+      </ul>
+
+      <p>While we are in pre-production, our infrastructure cannot (yet!) support all of the traffic generated by our users running <code>pip install</code>. We therefore ask that you do not use this version of PyPI for production installations.</p>
+      <p>On the other hand, uploads are working well and we encourage project maintainers and owners to use this version of PyPI over the legacy site.</p>
```
One slight tweak here, not sure if it matters but more than encouraging people to upload using Warehouse, we *require* it. Uploading to legacy no longer functions. Not sure if it makes sense to change this wording or not.

nlhkabu replied:
Proposed new text:

`On the other hand, uploads are working well and project maintainers and owners should use this version of PyPI over the legacy site, which no longer supports this function.`

is this ok @dstufft ?

dstufft replied:
Yup, sounds good.

nlhkabu commented:
Cleaned up PR commit history, and updated text based on feedback:

![screencapture-localhost-help-1513705521862](https://user-images.githubusercontent.com/3323703/34180077-44a040fc-e505-11e7-99fd-4dd8901b8614.png)

nlhkabu commented:
ping @dstufft for final review.  If you're happy I'll rebase and merge :)
